### PR TITLE
Switch back to `available_parallelism`

### PR DESCRIPTION
**Pull request by joshtriplett**
This reverts commit 8345cf5037506b38457483429fb113322c58f668 now that https://github.com/rust-lang/rust/pull/97925 is merged.

Since that
time, there are now multiple calls to get the number of CPUs, to handle
the `-j -NUM` case, so factor out a helper function.

### CODE REVIEW

File: src/cargo/core/compiler/build_config.rs [hunk 1]
```diff
@@ -1,11 +1,12 @@
 use crate::core::compiler::CompileKind;
 use crate::util::interning::InternedString;
 use crate::util::{CargoResult, Config, RustfixDiagnosticServer};
-use anyhow::bail;
+use anyhow::{bail, Context as _};
 use cargo_util::ProcessBuilder;
 use serde::ser;
 use std::cell::RefCell;
 use std::path::PathBuf;
+use std::thread::available_parallelism;
```
Comment (epage) [inline src/cargo/core/compiler/build_config.rs:9]: What makes this safe to do now compared to the concerns in https://github.com/rust-lang/cargo/pull/10737?

Comment (Muscraft) [reply src/cargo/core/compiler/build_config.rs:9]: It looks like rust-lang/rust#97925 fixed the issue with `available_parallelism` and cgroups v1 on Linux

Comment (epage) [reply src/cargo/core/compiler/build_config.rs:9]: I've updated the PR message to call that out.

**Review by epage — APPROVED** [review]

**Pull request merged**